spec/system/admin/products_v3/products_spec.rb

```diff
@@ -337,6 +337,62 @@
     end
   end
 
+  describe "Cloning Feature" do
+    let!(:variant_a1) {
+      create(:variant,
+             product: product_a,
+             display_name: "Medium box",
+             sku: "APL-01",
+             price: 5.25)
+    }
+    let(:product_a) { create(:simple_product, name: "Apples", sku: "APL-00") }
+
+    before do
+      visit admin_products_url
+    end
+
+    describe "Actions columns (clone)" do
+      it "shows an actions menu with a clone link when clicking on icon for product" do
+        within row_containing_name("Apples") do
+          page.find(".vertical-ellipsis-menu").click
+          expect(page).to have_link "Clone", href: spree.clone_admin_product_path(product_a)
+        end
+      end
+
+      it "does not show actions menu with a clone link when clicking on icon for variant" do
+        within row_containing_name("Medium box") do
+          page.find(".vertical-ellipsis-menu").click
+          expect(page).to_not have_link "Clone", href: spree.clone_admin_product_path(product_a)
+        end
+      end
```

Comment:
I know it's best practice to specify each expectation separately, but it can take a lot of resources to set up the test case each time. As these two don't affect each other, I would suggest putting them together to save time.

```suggestion
      it "shows an actions menu with a clone link when clicking on icon for product" do
        within row_containing_name("Apples") do
          page.find(".vertical-ellipsis-menu").click
          expect(page).to have_link "Clone", href: spree.clone_admin_product_path(product_a)
        end

        within row_containing_name("Medium box") do
          page.find(".vertical-ellipsis-menu").click
          expect(page).to_not have_link "Clone", href: spree.clone_admin_product_path(product_a)
        end
      end
```